api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -101,4 +115,16 @@ public void setEndDate(Date endDate) {
 	public int compareTo(CohortMembership o) {
 		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
 	}
+	
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof CohortMembership) {
+			CohortMembership o = (CohortMembership) obj;
+			return Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId()) &&
+					Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId()) &&
+					this.getStartDate() == o.getStartDate() &&
```

Comment:
You also need this to be consistent with equality semantics in OpenMRS, i.e if 2 CohortMemberships have the same uuid, they should be considered equal, basically you should delegate to super.equals first, if it returns false then you proceed to your code to further compare the memberships